metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageClient.scala

```diff
@@ -42,9 +42,18 @@ trait MetalsLanguageClient
   @JsonNotification("metals/executeClientCommand")
   def metalsExecuteClientCommand(params: ExecuteCommandParams): Unit
 
-  final def refreshModel(): Unit = {
-    val params = ClientCommands.RefreshModel.toExecuteCommandParams()
-    metalsExecuteClientCommand(params)
+  final def refreshModel(
+      codeLensRefreshSupport: Boolean
+  ): CompletableFuture[Void] = {
+    if (codeLensRefreshSupport) this.refreshCodeLenses
+    else {
+      val params = ClientCommands.RefreshModel.toExecuteCommandParams()
+      CompletableFuture.runAsync(
```

Comment:
I didn't find a better way to go from `Unit` to `CompletableFuture[Void]`

Reply:
That works but the future isn't really completed. There is still an command to send, no ?

Reply:
`metalsExecuteClientCommand` returns Unit so it seems more like a notification/fire & forget semantics. 
@tgodzik your attention will be needed here :D